### PR TITLE
Add TSA bundle annotation

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: Hector Fernandez <hector@chainguard.dev>

#### Summary

This change is related to https://github.com/sigstore/cosign/issues/2331. This PR adds a new annotation for the TSA Bundle.

### CODE REVIEW

File: pkg/oci/internal/signature/layer.go [hunk 1]
```diff
@@ -29,10 +29,11 @@ import (
...
 const (
-	sigkey    = "dev.cosignproject.cosign/signature"
-	certkey   = "dev.sigstore.cosign/certificate"
-	chainkey  = "dev.sigstore.cosign/chain"
-	BundleKey = "dev.sigstore.cosign/bundle"
+	sigkey       = "dev.cosignproject.cosign/signature"
+	certkey      = "dev.sigstore.cosign/certificate"
+	chainkey     = "dev.sigstore.cosign/chain"
+	BundleKey    = "dev.sigstore.cosign/bundle"
+	TSABundleKey = "dev.sigstore.cosign/bundle"
```
Comment (priyawadhwa) [inline pkg/oci/internal/signature/layer.go:36]: I think you might have copy/pasted BundleKey here!

**Review by priyawadhwa — COMMENTED** [review]
nice! just left one comment.

**Review by haydentherapper — COMMENTED** [review]
Nice work!

File: pkg/oci/internal/signature/layer.go (hunk 1 above)
Comment (hectorj2f) [reply pkg/oci/internal/signature/layer.go:36]: Good catch! I named here as `tsabundle` https://github.com/sigstore/cosign/pull/2454/files#diff-a4859b65c42ca40ac004766c29a8b1279097c2d7909552f77885af38876729c8R36. But I am not good at naming, so we can use 3161timestamp instead.

**Review by priyawadhwa — APPROVED** [review]
lgtm, tests looks great!

**Pull request merged**